### PR TITLE
Fixed matrix verifier to actually work

### PR DESCRIPTION
**Pull request by etwest**
Title 

Also I'm going to add some time hooks.

Also also, some small changes to the way we test sketches. We expect a sketch failure rate of 1/100 and the tests did not take this into account

### CODE REVIEW

File: include/test/mat_graph_verifier.h [hunk 1]
```diff
@@ -12,43 +12,24 @@
 class MatGraphVerifier : public GraphVerifier {
   std::vector<std::set<node_id_t>> kruskal_ref;
   std::vector<std::set<node_id_t>> boruvka_cc;
-  std::vector<bool>& det_graph;
+  std::vector<std::vector<bool>> adj_graph;
```
Comment (Victor-C-Zhang) [inline include/test/mat_graph_verifier.h:15]: is this the fix?

**Review by Victor-C-Zhang — COMMENTED** [review]
If having this merged isn't critical for submission, we should merge this and the time_hook branch

File: include/test/mat_graph_verifier.h (hunk 1 above)
Comment (etwest) [reply include/test/mat_graph_verifier.h:15]: It is most of it. The idea is to avoid the edge_id map and just use an adjacency matrix.

**Comment by etwest** [comment]
> If having this merged isn't critical for submission, we should merge this and the time_hook branch

What's in there that we want?

**Comment by Victor-C-Zhang** [comment]
> > If having this merged isn't critical for submission, we should merge this and the time_hook branch
> 
> What's in there that we want?

A lot of the same time hooks you have that the current experiments rely on

**Comment by etwest** [comment]
> > > If having this merged isn't critical for submission, we should merge this and the time_hook branch
> > 
> > 
> > What's in there that we want?
> 
> A lot of the same time hooks you have that the current experiments rely on

I've got those in here as well. Can we merge this and then open another pull request for anything we want from the time_hook branch?

**Review by Victor-C-Zhang — APPROVED** [review]

**Pull request merged**